src/util/misc/isTransparent.ts

```diff
@@ -28,24 +29,19 @@ export const isTransparent = (
     }
   }
 
-  let _isTransparent = true;
   const { data } = ctx.getImageData(
     x,
     y,
     tolerance * 2 || 1,
     tolerance * 2 || 1
   );
```

Comment:
```suggestion
  const { data } = ctx.getImageData(
    x - tolerance,
    y - tolerance,
    tolerance * 2 + 1,
    tolerance * 2 + 1,
  );
```

If we write this in this way we can get rid of  the if condition above.
Reading getImageData out of bounds seems in spec

```
The getImageData(sx, sy, sw, sh, settings) method steps are:

If either the sw or sh arguments are zero, then throw an ["IndexSizeError"](https://webidl.spec.whatwg.org/#indexsizeerror) [DOMException](https://webidl.spec.whatwg.org/#dfn-DOMException).

If the [CanvasRenderingContext2D](https://html.spec.whatwg.org/multipage/canvas.html#canvasrenderingcontext2d)'s [origin-clean](https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-origin-clean) flag is set to false, then throw a ["SecurityError"](https://webidl.spec.whatwg.org/#securityerror) [DOMException](https://webidl.spec.whatwg.org/#dfn-DOMException).

Let imageData be a [new](https://webidl.spec.whatwg.org/#new) [ImageData](https://html.spec.whatwg.org/multipage/canvas.html#imagedata) object.

[Initialize](https://html.spec.whatwg.org/multipage/canvas.html#initialize-an-imagedata-object) imageData given sw, sh, [settings](https://html.spec.whatwg.org/multipage/canvas.html#initialize-imagedata-settings) set to settings, and [defaultColorSpace](https://html.spec.whatwg.org/multipage/canvas.html#initialize-imagedata-defaultcolorspace) set to [this](https://webidl.spec.whatwg.org/#this)'s [color space](https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-color-space).

Let the source rectangle be the rectangle whose corners are the four points (sx, sy), (sx+sw, sy), (sx+sw, sy+sh), (sx, sy+sh).

Set the pixel values of imageData to be the pixels of [this](https://webidl.spec.whatwg.org/#this)'s [output bitmap](https://html.spec.whatwg.org/multipage/canvas.html#output-bitmap) in the area specified by the source rectangle in the bitmap's coordinate space units, converted from [this](https://webidl.spec.whatwg.org/#this)'s [color space](https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-color-space) to imageData's [colorSpace](https://html.spec.whatwg.org/multipage/canvas.html#dom-imagedata-colorspace) using ['relative-colorimetric'](https://drafts.csswg.org/css-color/#valdef-color-profile-rendering-intent-relative-colorimetric) rendering intent.

Set the pixels values of imageData for areas of the source rectangle that are outside of the [output bitmap](https://html.spec.whatwg.org/multipage/canvas.html#output-bitmap) to [transparent black](https://drafts.csswg.org/css-color/#transparent-black).

Return imageData.
```

Reply:
- `document.createElement('canvas').getContext('2d').getImageData(0,0,1,1)` returns [0,0,0,0] though IMO it should return null

Reply:
so that is fine as well

Reply:
getImageData(0,0,1,1) should give you a pixel correct? why would you have null?

Reply:
I think the spec is confusing. I am asking for a non existing pixel and I get one. But for our sake it is fine since it is transparent

Reply:
oh i see what you mean.
I think is a point of view. Having always pixels is definitely more ergonomic.
Of course if you are hunting for transparent pixels and you go accidentally out of bounds, you are confused.